Fails with error 'Only non-capturing groups are accepted: e.g. (?:pattern) instead of (pattern)'

Building and running the current head, the container fails to start with error:

```
2017/01/18 21:07:08 Starting Portainer on :9000
panic: route /upload/tls/{endpointID}/{certificate:(ca|cert|key)} contains capture groups in its regexp. Only non-capturing groups are accepted: e.g. (?:pattern) instead of (pattern)

goroutine 1 [running]:
panic(0x79aec0, 0xc420251fb0)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/gorilla/mux.newRouteRegexp(0x840e01, 0x34, 0x0, 0x7fc87d2d0000, 0x0, 0x70)
        /go/src/github.com/gorilla/mux/regexp.go:116 +0xe1b
github.com/gorilla/mux.(*Route).addRegexpMatcher(0xc42025d9d0, 0x840e01, 0x34, 0x0, 0x1, 0xc42025d9d0)
        /go/src/github.com/gorilla/mux/route.go:163 +0xe1
github.com/gorilla/mux.(*Route).Path(0xc42025d9d0, 0x840e01, 0x34, 0x0)
        /go/src/github.com/gorilla/mux/route.go:327 +0x4f
github.com/gorilla/mux.(*Router).Handle(0xc420254be0, 0x840e01, 0x34, 0xa07d00, 0xc420253020, 0xc420250e00)
        /go/src/github.com/gorilla/mux/mux.go:220 +0xf3
github.com/portainer/portainer/http.NewUploadHandler(0xc420221310, 0xc420240a00)
        /go/src/github.com/portainer/portainer/http/upload_handler.go:31 +0x264
github.com/portainer/portainer/http.(*Server).Start(0xc420161220, 0x18, 0xc4201a3e78)
        /go/src/github.com/portainer/portainer/http/server.go:64 +0x267
main.main()
        /src/cmd/portainer/main.go:88 +0x58c

```

Attached is a Vagrantfile that reproduces this.


[Vagrantfile.txt](https://github.com/portainer/portainer/files/715083/Vagrantfile.txt)

--- a/api/http/upload_handler.go
+++ b/api/http/upload_handler.go
@@ -26,7 +26,7 @@ func NewUploadHandler(middleWareService *middleWareService) *UploadHandler {
 		Logger:            log.New(os.Stderr, "", log.LstdFlags),
 		middleWareService: middleWareService,
 	}
-	h.Handle("/upload/tls/{endpointID}/{certificate:(ca|cert|key)}", middleWareService.addMiddleWares(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	h.Handle("/upload/tls/{endpointID}/{certificate:(?:ca|cert|key)}", middleWareService.addMiddleWares(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		h.handlePostUploadTLS(w, r)
 	})))
 	return h

